ml-agents/mlagents/trainers/tests/test_ppo.py

```diff
@@ -499,5 +501,41 @@ def test_normalization(dummy_config):
     assert (variance[0] - 1) / steps == pytest.approx(0.152, abs=0.01)
 
 
+def test_min_visual_size():
+    # Make sure each EncoderType has an entry in MIS_RESOLUTION_FOR_ENCODER
+    assert set(LearningModel.MIS_RESOLUTION_FOR_ENCODER.keys()) == set(EncoderType)
+
+    for encoder_type in EncoderType:
+        with tf.Graph().as_default():
+            good_size = LearningModel.MIS_RESOLUTION_FOR_ENCODER[encoder_type]
+            good_res = CameraResolution(
+                width=good_size, height=good_size, num_channels=3
+            )
+            LearningModel._check_resolution_for_encoder(good_res, encoder_type)
+            vis_input = LearningModel.create_visual_input(
+                good_res, "test_min_visual_size"
+            )
+            enc_func = LearningModel.get_encoder_for_type(encoder_type)
+            enc_func(vis_input, 32, LearningModel.swish, 1, "test", False)
+
+        # Anything under the min size should raise an exception. If not, decrease the min size!
+        with pytest.raises(Exception):
```

Comment:
Should we except the specific exception (and check that it's the negative dimension issue) just in case the model has a different error?

Reply:
I don't think we need the exact type; the test is sufficient to make sure that size N works and N-1 doesn't. Besides, it's possible that different encoders would fail for different reasons.